### ISSUE TITLE
[BUG] Bundle() does not restore CWD call when baseDir is used

### ISSUE THREAD

**Issue by g-radam** [asyncapi/bundler#166]
### Describe the bug.

Invoking the `bundle()` function with a `baseDir` option provided will have the working directory changed, but will not be changed back on function exit. Successive calls to this method where each call provides a _relative_ path `baseDir` option will fail, specifically on the 2nd call, since the original relative path no longer makes sense in the CWD.

Error:
```
Error: ENOENT: no such file or directory, chdir '<project>/src' -> './src'
    at process.wrappedChdir (node:internal/bootstrap/switches/does_own_process_state:130:14)
    at process.chdir (<project>/node_modules/graceful-fs/polyfills.js:22:11)
    at bundle (<project>/node_modules/@asyncapi/bundler/lib/index.js:85:17)
    at build (<project>/gulpfile.js:37:30)
```

The lines in question are:
 - https://github.com/asyncapi/bundler/blob/c9ae036692c55964d56b0f4948525b2a2ffbe1c9/src/index.ts#L90
 - https://github.com/asyncapi/bundler/blob/c9ae036692c55964d56b0f4948525b2a2ffbe1c9/src/index.ts#L92

The function will need to copy the CWD, and restore it prior to exiting. In cases where the function body throws, the exception should most likely be caught, the CWD reverted, and the exception be propagated as per normal.

### Expected behavior

Invoking the bundle restores the CWD.

### Screenshots

N/A

### How to Reproduce

```
  // Bundle server AsyncAPI files
  const serverMerged = await bundle(["common.yaml"], {
    base: "server.yaml",
    baseDir: "./src/",
    xOrigin: false,
  });

  // Merge client AsyncAPI files
  // Fails here due to ./src/ relative path no longer correct, as the cwd IS NOW ./src/. 
  const clientMerged = await bundle(["common.yaml"], {
    base: "client.yaml",
    baseDir: "./src/",
    xOrigin: false,
  });
```

### 🥦 Browser

None

### 👀 Have you checked for similar open issues?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

**Comment by aeworxet** [comment]
Please copy&paste this compiled `index.js` to your `index.js` and check if it solves the issue.

<details>
<summary>index.js</summary>

```js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.originDir = void 0;
const fs_1 = require("fs");
const path_1 = __importDefault(require("path"));
const util_1 = require("./util");
const document_1 = require("./document");
const parser_1 = require("./parser");
exports.originDir = String(process.env.PWD);
async function bundle(files, options = {}) {
    // if one string was passed, convert it to an array
    if (typeof files === 'string') {
        files = Array.from(files.split(' '));
    }
    if (options.baseDir && typeof options.baseDir === 'string') {
        process.chdir(path_1.default.resolve(exports.originDir, options.baseDir));
    }
    else if (options.baseDir && Array.isArray(options.baseDir)) {
        process.chdir(path_1.default.resolve(exports.originDir, String(options.baseDir[0]))); // guard against passing an array
    }
    const readFiles = files.map(file => (0, fs_1.readFileSync)(file, 'utf-8')); // eslint-disable-line
    const parsedJsons = readFiles.map(file => (0, util_1.toJS)(file));
    const majorVersion = (0, util_1.versionCheck)(parsedJsons);
    if (typeof options.base !== 'undefined') {
        if (typeof options.base === 'string') {
            options.base = (0, fs_1.readFileSync)(options.base, 'utf-8'); // eslint-disable-line
        }
        else if (Array.isArray(options.base)) {
            options.base = (0, fs_1.readFileSync)(String(options.base[0]), 'utf-8'); // eslint-disable-line
        }
        options.base = (0, util_1.toJS)(options.base);
        await (0, parser_1.parse)(options.base, majorVersion, options);
    }
    const resolvedJsons = await (0, util_1.resolve)(parsedJsons, majorVersion, options);
    process.chdir(exports.originDir);
    return new document_1.Document(resolvedJsons, options.base);
}
exports.default = bundle;
// 'module.exports' is added to maintain backward compatibility with Node.js
// projects, that use CJS module system.
module.exports = bundle;
```

</details>

**Comment by g-radam** [comment]
I can confirm that worked :)

**Comment by aeworxet** [comment]
@g-radam
This change is [included in v0.5.2](https://github.com/asyncapi/bundler/pull/167#issuecomment-2133125873). Please check once again with the published version.

**Comment by g-radam** [comment]
@aeworxet  I've validated that the issue is fixed in v0.5.2! :) Many thanks aeworxet!
